### PR TITLE
chore(sonar): cleanup 8 — clear 11 smells, cut duplication, raise coverage

### PR DESCRIPTION
**Pull request by asafgolombek**
## Summary

SonarCloud hygiene pass (cleanup 8). The quality gate on `main` was already **PASS** (0 bugs, 0 vulns, 0 unreviewed hotspots) — this clears the 11 outstanding `CODE_SMELL` findings, reduces duplication, and raises coverage on the most tractable partial file. Fix-not-exclude per repo convention.

## Issues cleared (11)
- **3× S3776** (cognitive complexity): `parseNimbusConnectorsToml` → `accumulateConnectorTables` + `resolveConnectorConfig`; `assemblePlatformServices` → `buildTeamCredentialContexts` + `bootChatopsIntoAssembly` (the late-bound `identityBootRef` became a holder; the two duplicated identity spreads collapsed into one); `aggregateContributions` → `collectPeerItems`
- **2× S6582**: optional-chain in `invoke-gate.ts` (I19/I26 load-bearing — behavior identical, invariants preserved)
- **2× S7747**: dropped redundant array-spread in `embedding-worker-core` `idle()`
- **2× S6353**: `\w` over `[A-Za-z0-9_]` in `format-audit-payload` (`_`-boundary semantics preserved)
- **S4325**: removed redundant `BonjourLike` type assertion
- **S7781**: `replaceAll` over `replace(/'/g)` in snowflake `sfLiteral`

## Duplication
- Hoisted the byte-identical GitHub Actions I/O helpers (`safeString`/`safeInt`/`getInput`/`getBooleanInput`/`getIntInput`/`writeJobSummary`/`emitAnnotation`) plus a `makeSetOutput(allowedNames)` factory into a new `packages/github-actions/shared/gha-io.ts`. `preflight-query` + `annotate-action` import and re-export what their tests need; dist bundles rebuilt. Kills the 83%/81% `output.ts` and the `main.ts` scaffolding twins.
- `monte-carlo/search-filter` now uses the shared `fieldsFromKeys` (drops its duplicate `stringAt`/`fieldsOf`), matching the bitrise/codemagic pattern.

## Coverage
- `connector.ts` **91.2% → 95.6%**: real behavior tests for the `relTime`/`fmtNextSync` time buckets, `truncateText`, `fmtHealthRetry`, and flag-value validation edges.
- `assemble-sync-registrations.ts` (53%) intentionally **not** chased — it is explicitly excluded from the coverage floor as boot wiring-glue, and covering its ~95 connector closures would require running 95 real sync paths for zero gate value.

## Validation
- Full sequential typecheck (all packages) ✓
- Gateway full suite (10006 pass) + CLI (1693) + monte-carlo + gha (55) ✓
- Security-invariant structure audit (I19/I22/I25) ✓
- **Docker Linux-authoritative coverage-floor: ok** (1002 files scanned) ✓
- Biome (`bunx biome check packages scripts`, 2765 files) ✓
- Independent code review of the diff: no blocking issues ✓

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Refactor**
  * Centralized GitHub Actions input/output, annotations, and job summaries into a shared module; updated related actions to re-use shared helpers.
  * Refined Nimbus TOML connector parsing/validation and improved gateway contribution aggregation; extracted helpers across related query and assembly flows.
* **Bug Fixes**
  * Tightened fine-grained GitHub credential redaction patterns and improved Snowflake single-quote escaping.
* **Tests**
  * Expanded CLI list/auth edge-case tests (time/health buckets, formatting, truncation) and added shared action utility unit tests.
* **Documentation / CI**
  * Updated link checking to use an API token and improved CI coverage for shared action utilities.

### CODE REVIEW

File: packages/github-actions/shared/gha-io.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const file = process.env.GITHUB_STEP_SUMMARY;
+  if (file === undefined) return;
+  const safe = md.length > STEP_SUMMARY_MAX_BYTES ? md.slice(0, STEP_SUMMARY_MAX_BYTES) : md;
+  appendFileSync(file, `${safe}\n`);
+}
+
+/** Emit a GitHub Actions workflow annotation command on stdout. */
+export function emitAnnotation(level: "warning" | "error", message: string): void {
+  process.stdout.write(`::${level}::${message}\n`);
+}
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/github-actions/shared/gha-io.ts]: Fixed in 2d8303f7. `emitAnnotation` now escapes the message with GitHub's documented data encoding (`%`→`%25`, CR→`%0D`, LF→`%0A`), so gateway-supplied content with newlines or `::` can no longer inject a second workflow command; GitHub decodes it back when rendering, so multi-line annotations still display. Added `gha-io.test.ts` with an explicit injection test (and wired `github-actions/shared` into the CI test loop, mirroring `mcp-connectors/shared`).

File: packages/github-actions/shared/gha-io.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    written = "";
+  });
+
+  function capture(): void {
+    written = "";
+    // biome-ignore lint/suspicious/noExplicitAny: minimal stdout.write spy for the test
+    process.stdout.write = ((chunk: any) => {
+      written += String(chunk);
+      return true;
+    }) as typeof process.stdout.write;
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/github-actions/shared/gha-io.test.ts]: Fixed in 7e630ee1. Replaced the `any` (and its biome-ignore) with `Parameters<typeof process.stdout.write>[0]` (`string | Uint8Array`), decoding non-string chunks via `Buffer.from`. Satisfies the No-`any` non-negotiable; 8 tests still pass, biome clean.
